controllers/build.go

```diff
@@ -0,0 +1,282 @@
+func paramsForInitialBuild(component appstudiov1alpha1.Component) []tektonapi.Param {
+	sourceCode := component.Spec.Source.GitSource.URL
+	outputImage := component.Spec.Build.ContainerImage
+
+	params := []tektonapi.Param{
+		{
+			Name: "git-url",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: sourceCode,
+			},
+		},
+		{
+			Name: "output-image",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: outputImage,
+			},
+		},
+	}
+
+	return params
+}
+
+// paramsForWebhookBasedBuilds returns the 'input' paramters for the webhook-triggered
+// PipelineRuns. The key difference between webhook triggered PipelineRuns and user-triggered
+// PipelineRuns would be that you'd have the git revision appended to the output image tag
+func paramsForWebhookBasedBuilds(component appstudiov1alpha1.Component) []tektonapi.Param {
+	sourceCode := component.Spec.Source.GitSource.URL
+	outputImage := normalizeOutputImageURL(component.Spec.Build.ContainerImage)
+
+	params := []tektonapi.Param{
+		{
+			Name: "git-url",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: sourceCode,
+			},
+		},
+		{
+			Name: "output-image",
+			Value: tektonapi.ArrayOrString{
+				Type:      tektonapi.ParamTypeString,
+				StringVal: outputImage,
+			},
+		},
+	}
+
+	return params
+}
+
+func commonLabels(component *appstudiov1alpha1.Component) map[string]string {
+	labels := map[string]string{
+		"build.appstudio.openshift.io/build":       "true",
+		"build.appstudio.openshift.io/type":        "build",
+		"build.appstudio.openshift.io/version":     "0.1",
+		"build.appstudio.openshift.io/component":   component.Name,
+		"build.appstudio.openshift.io/application": component.Spec.Application,
+	}
+	return labels
+}
+
+// commonStorage returns the PVC that would be created per namespace for
+// user-triggered and webhook-triggered Tekton workspaces.
+func commonStorage(component appstudiov1alpha1.Component, name string) corev1.PersistentVolumeClaim {
+	fsMode := corev1.PersistentVolumeFilesystem
+
+	workspaceStorage := &corev1.PersistentVolumeClaim{
+		ObjectMeta: v1.ObjectMeta{
+			Name:        name,
+			Namespace:   component.Namespace,
+			Annotations: commonLabels(&component),
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				"ReadWriteOnce",
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					"storage": resource.MustParse("10Mi"),
+				},
+			},
+			VolumeMode: &fsMode,
+		},
+	}
+	return *workspaceStorage
+}
+
+// route returns the Route resource that would enable
+// ingress traffic into the webhook endpoint ( aka EventListener)
+// TODO: This needs to be secure.
+func route(component appstudiov1alpha1.Component) routev1.Route {
+	var port int32 = 8080
+	webhook := &routev1.Route{
+		ObjectMeta: v1.ObjectMeta{
+			Name:        component.Name,
```

Comment:
Should we use el-" + component.Name for the route name here? We also use the component name for the route that we generate in the gitops repository (see https://github.com/redhat-appstudio/application-service/blob/main/gitops/generate.go#L170), so there would be a conflict between these two routes

Reply:
Good point, let me rename that..

Reply:
Done f3d6037